ATMoviess/ATMoviess/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,87 @@
+﻿using ATMoviess.Models;
+using ATMoviess.Services;
+using ATMoviess.Services.Navigation;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace ATMoviess.ViewModels
+{
+    public class MainPageViewModel : ViewModelBase
+    {
+        INavigationService navigation;
+        MoviesService moviesService;
+
+        public MainPageViewModel(INavigationService _navigation)
+        {
+            navigation = _navigation;
+            NavigateToDetailsCommand = new Command(NavigateToDetails);
+            SearchMovieTitleCommand = new Command(SearchMovieTitle);
+        }
+
+        public async override Task InitializeAsync(object navigationData)
+        {
+            IsLoading = true;
+
+            moviesService = new MoviesService();
```

Comment:
As you are using an `IoC` container, you could just inject the service in the `ViewModel's` constructor.

Reply:
That's true. Thanks for pointing that out!